QMLComponents/controls/variableslistbase.cpp

```diff
@@ -338,59 +340,63 @@ void VariablesListBase::termsChangedHandler()
 }
 
 
-int VariablesListBase::_getAllowedColumnsTypes()
+int VariablesListBase::_getColumnsTypes(const QStringList& types)
 {
-	int allowedColumnsTypes = -1;
+	int columnsTypes = -1;
 
-	if (!allowedColumns().isEmpty())
+	if (!types.isEmpty())
 	{
-		allowedColumnsTypes = 0;
-		for (const QString& allowedColumn: allowedColumns())
+		columnsTypes = 0;
+		for (const QString& typeStr: types)
 		{
-			columnType allowedType = columnTypeFromString(fq(allowedColumn), columnType::unknown);
-			if (allowedType != columnType::unknown)
-				allowedColumnsTypes |= int(allowedType);
+			columnType type = columnTypeFromString(fq(typeStr), columnType::unknown);
+			if (type != columnType::unknown)
```

Comment:
This check is kind of useless as `unknown` == 0 anyway. Although I guess it does make sense for the error/warning underneath.